pkg/detectors/postgres/postgres.go

```diff
@@ -0,0 +1,170 @@
+package postgres
+
+import (
+	"context"
+    "database/sql"
+	"net/url"
+	"regexp"
+	"strings"
+    "fmt"
+    "time"
+
+    "github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	// URI pattern for PostgreSQL connection string
+	uriPat = regexp.MustCompile(`\b(?i)postgresql://[\S]+\b`)
+
+	// Separate patterns for username, password, and hostname
+    hostnamePat = regexp.MustCompile(`(?i)(?:host|server).{0,40}?(\b[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*\b)`)
+
+
+	// You might want to customize these patterns based on common practices in your codebases
+)
+
+func (s Scanner) Keywords() []string {
+	return []string{"postgres", "psql", "pghost"}
+}
+
+func verifyPostgres(pgURL *url.URL) error {
+    // Extract the necessary components
+    username := ""
+    password := ""
+    if pgURL.User != nil {
+        username = pgURL.User.Username()
+        password, _ = pgURL.User.Password()
+    }
+    hostname := pgURL.Hostname()
+
+    // Handle custom port
+    port := pgURL.Port()
+    if port == "" {
+        port = "5432" // Default PostgreSQL port
+    }
+
+    // Handle SSL mode
+    sslmode := "disable" // Default to disable
+    queryParams := pgURL.Query()
+    if sslQuery, ok := queryParams["sslmode"]; ok && len(sslQuery) > 0 {
+        sslmode = sslQuery[0]
+    }
+
+    // Construct the PostgreSQL connection string
+    connStr := fmt.Sprintf("user=%s password=%s host=%s port=%s sslmode=%s", username, password, hostname, port, sslmode)
+
+    // Open a connection to the database
+    db, err := sql.Open("postgres", connStr)
+    if err != nil {
+        return err
+    }
+    defer db.Close()
+
+    ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
Is there support for providing default time.Durations and allowing this to be configured at runtime with the cli?

Reply:
No, but that would be a useful enhancement across detectors. I find that 3s for private key lookup can easily time out if there's a lot of network traffic.